util/tls/tls.go

```diff
@@ -144,7 +147,23 @@ func GetServerTLSConfigFromSecret(ctx context.Context, kubectlConfig kubernetes.
 		return nil, err
 	}
 
+	capem, err := util.GetSecrets(ctx, kubectlConfig, namespace, tlsKubernetesSecretName, tlsCaSecretKey)
+	if err != nil {
+		return nil, err
+	}
+
+	rootCAs, err := x509.SystemCertPool()
+	if rootCAs != nil {
+		log.Warnf("failed to get system certificate pool: %v, continuing with empty certificate trust", err)
+		rootCAs = x509.NewCertPool()
+	}
+
+	if !rootCAs.AppendCertsFromPEM(capem) {
+		log.Warn("failed to append ca.crt to the trusted CA pool")
```

Comment:
Happy to make it one, but I'm not sure..
The x509.CertPool API doesn't really return any errors, so decided to follow suit?

Reply:
I don't know if this is an error or not

Reply:
Hmm. Well, if this fails, it will just use the standard Certificate trust store, without the local certificate being trusted.

If we make this an error, it means we need `ca.crt` to be a valid cert within the secret. Only warning on this means that the Cert secret could be a publicly trusted cert, and `ca.crt` doesn't need to be valid.

This makes me think, maybe we shouldn't expect `ca.crt` to always exist though - but rather just warn/info if it doesn't?

Happy to make any changes... lmk.

Reply:
I've just updated the logic a little bit here.
Secrets are no longer _required_ to contain a `ca.crt`, instead of erroring out, it will just warn that it skipped adding a `ca.crt` to the list of trusted certificates.

The system root CA's are still trusted in all cases, so publicly signed certificates will be fine. 
This should be more compatible with different setups.